test/general/sycl_policy.pass.cpp

```diff
@@ -50,20 +50,63 @@ void test_policy_instance(const Policy& policy)
-        sycl::queue{TestUtils::default_selector, sycl::property::queue::in_order()}));
-#endif
+    using namespace oneapi::dpl::execution;
+    auto q = sycl::queue{TestUtils::default_selector};
+
+    EXPECT_TRUE((is_execution_policy<device_policy<class Kernel_0>>::value), "wrong result for is_execution_policy<device_policy>");
+
+    test_policy_instance(dpcpp_default);
+
+    // make_device_policy
+    test_policy_instance(make_device_policy<class Kernel_11>(q));
+    test_policy_instance(make_device_policy<class Kernel_12>(TestUtils::default_selector));
+    test_policy_instance(make_device_policy<class Kernel_13>(sycl::device{TestUtils::default_selector}));
+    test_policy_instance(make_device_policy<class Kernel_14>(sycl::queue{TestUtils::default_selector, sycl::property::queue::in_order()}));
+    test_policy_instance(make_device_policy<class Kernel_15>(dpcpp_default));
+    test_policy_instance(make_device_policy<class Kernel_16>());
+
+    // device_policy
+    EXPECT_TRUE(device_policy<class Kernel_1>(q).queue() == q, "wrong result for queue()");
+    test_policy_instance(device_policy<class Kernel_21>(q));
+    test_policy_instance(device_policy<class Kernel_22>(sycl::device{TestUtils::default_selector}));
+    test_policy_instance(device_policy<class Kernel_23>(dpcpp_default));
+    test_policy_instance(device_policy<class Kernel_24>(sycl::queue(dpcpp_default))); // conversion to sycl::queue
+    test_policy_instance(device_policy<>{});
+    class Kernel_25;
+    EXPECT_TRUE((std::is_same<device_policy<Kernel_25>::kernel_name, Kernel_25>::value), "wrong result for kernel_name");
+
+#if ONEDPL_FPGA_DEVICE
+    EXPECT_TRUE((is_execution_policy<fpga_policy</*unroll_factor =*/ 1, class Kernel_0>>::value), "wrong result for is_execution_policy<fpga_policy>");
+    test_policy_instance(dpcpp_fpga);
+
+    // make_fpga_policy
+    test_policy_instance(make_fpga_policy</*unroll_factor =*/ 1, class Kernel_31>(sycl::queue{TestUtils::default_selector}));
+    test_policy_instance(make_fpga_policy</*unroll_factor =*/ 2, class Kernel_32>(sycl::device{TestUtils::default_selector}));
+    test_policy_instance(make_fpga_policy</*unroll_factor =*/ 4, class Kernel_33>(dpcpp_fpga));
+    test_policy_instance(make_fpga_policy</*unroll_factor =*/ 8, class Kernel_34>());
+    test_policy_instance(make_fpga_policy</*unroll_factor =*/ 16, class Kernel_35>(sycl::queue{TestUtils::default_selector}));
+
+    // fpga_policy
+    test_policy_instance(fpga_policy</*unroll_factor =*/ 1, class Kernel_41>(sycl::queue{TestUtils::default_selector}));
+    test_policy_instance(fpga_policy</*unroll_factor =*/ 2, class Kernel_42>(sycl::device{TestUtils::default_selector}));
+    test_policy_instance(fpga_policy</*unroll_factor =*/ 4, class Kernel_43>(dpcpp_fpga));
+    test_policy_instance(fpga_policy</*unroll_factor =*/ 8, class Kernel_44>{});
+    EXPECT_TRUE((fpga_policy</*unroll_factor =*/ 8, class Kernel_45>::unroll_factor == 8), "wrong unroll_factor");
```

Comment:
Let's add a check for `kernel_name` like it is done for `device_policy`. I guess this is omitted as `fpga_policy` is based on `device_policy`, but such implementation details should not affect tests.